src/cregistry/entry.c

```diff
@@ -517,16 +517,45 @@ int reg_entry_installed(reg_registry* reg, char* name, reg_entry*** entries,
  *
  * @param [in] reg     registry to search in
  * @param [in] path    path of the file to check ownership of
+ * @param [in] cs      false if check should be performed case-insensitive,
+ *                     true otherwise
  * @param [out] entry  the owner, or NULL if no active port owns the file
  * @param [out] errPtr on error, a description of the error that occurred
  * @return             true if success; false if failure
  */
-int reg_entry_owner(reg_registry* reg, char* path, reg_entry** entry,
+int reg_entry_owner(reg_registry* reg, char* path, int cs, reg_entry** entry,
         reg_error* errPtr) {
     int result = 0;
     sqlite3_stmt* stmt = NULL;
-    char* query = "SELECT id FROM registry.files WHERE actual_path=? AND active";
     int lower_bound = 0;
+    const char* base_query_start = "SELECT id FROM registry.files WHERE actual_path";
+    const char* base_query_mid = "?";
+    const char* base_query_end = " AND active";
+    const char* cis_operator = " LIKE ";
+    const char* cs_operator = "=";
+    const char* extend_query = " COLLATE NOCASE";
+    const int query_size = strlen(base_query_start) + strlen(cs_operator)
+                           + strlen(cis_operator) + strlen(base_query_mid)
+                           + strlen(base_query_end) + strlen(extend_query) + 1;
+    char* query = malloc(query_size);
+
+    if (!query) {
+        return 0;
+    }
+
+    strncpy(query, base_query_start, query_size - 1);
+    if (!cs) {
+        strncat(query, cis_operator, query_size - strlen(query) - 1);
+    }
+    else {
+        strncat(query, cs_operator, query_size - strlen(query) -1);
+    }
+    strncat(query, base_query_mid, query_size - strlen(query) - 1);
+    strncat(query, base_query_end, query_size - strlen(query) - 1);
+    if (!cs) {
+        strncat(query, extend_query, query_size - strlen(query) - 1);
+    }
```

Comment:
Well great, I had a comment here that somehow vanished.

I don't like the `strncat` funny business. Wouldn't using `snprintf` or `asprintf` be much shorter and more readable?

Consider:
```c
char *query;
if (-1 == asprintf(&ret, "SELECT id FROM registry.files WHERE actual_path %s ? AND active", cs ? "=" : "LIKE")) {
    return 0;
}

/* or with COLLATE NOCASE */
asprintf(&ret, "SELECT id FROM registry.files WHERE actual_path %s ? AND active %s", cs ? "=" : "LIKE", cs ? "" : "COLLATE NOCASE");

/* sqlite3_prepare_v2(...) */

free(query);
```

Or, if you want to avoid the non-standard `asprintf` (even though I think we already use it in MacPorts) or the `malloc`:
```c
char query[strlen("SELECT id FROM registry.files WHERE actual_path LIKE ? AND active COLLATE NOCASE") + 1];
snprintf(query, sizeof(query), "SELECT id FROM registry.files WHERE actual_path %s ? AND active %s",
         cs ? "=" : "LIKE", cs ? "" : "COLLATE NOCASE");
```